hatchery/nextflow.go

```diff
@@ -1075,11 +1082,52 @@ func setupSubnet(subnetName string, cidr string, vpcid string, ec2Svc *ec2.EC2)
 		return exsubnet.Subnets[0].SubnetId, nil
 	}
 
+	// Fetch all availability zones
+	// this is being limited to a region by the ec2svc that gets passed in.
+	describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe availability zones: %v", err)
+	}
+
+	var selectedZone string
+	for _, zone := range describeZonesOutput.AvailabilityZones {
+		if *zone.State == "available" {
+			input := &ec2.DescribeInstanceTypeOfferingsInput{
+				LocationType: aws.String("availability-zone"),
+				Filters: []*ec2.Filter{
```

Comment:
I think it is better for the sake of posterity,  if we add a comment line here indicating that some AZs lack the compute infrastructure for our needs (citing the links posted in the other [PR](https://github.com/uc-cdis/hatchery/pull/110)'s description. 

Just so as to not wonder why we wanted to filter on this particularly.

Reply:
If you think,  adding that in code could seem more confusing in the future, maybe we can do that in this PR's description

Reply:
Added comment 🙌